src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -483,6 +487,11 @@ public Void caseStringType() {
 
         @Override
         public Void caseObjectType(ObjectType type) {
+          // Closure doesn't require that all the type params be declared, but TS does
```

Comment:
added a test case for that, and split these tests from `types.js` into new `generics.js`